### PR TITLE
Fix/task reassign and assign all on date bug

### PR DESCRIPTION
**Pull request by fsayyid**
Ticket: [LF-2323](https://lite-farm.atlassian.net/browse/LF-2323) (see most recent comment)

This pull request was made to allow users to reassign a task to a user and assign all unassigned tasks to the same user when "assign all unassigned tasks on this date to this person" is checked. 

**Summary of Changes:** 
- Refactored assignTask endpoint so that logic can be shared with assignAllTasksOnDate endpoint
- Made it so that reassigning a task is possible from assignAllTasksOnDate endpoint
- Added tests for this case

**To Test:**
- Run backend tests in task.test.js, make sure they pass
- Have at least two users at a farm
- Create a few tasks (maybe about 4) with at least two being on the same date
- Assign one of the tasks on the same date to a user
- Make sure that another task on the same date is unassigned 
- Reassign that task to a different user while having "assign all unassigned tasks on this date to this person" checked
- Ensure that the previously unassigned tasks on the same date and the task you reassigned are all assigned to the intended user
- Play around with assigning, unassigning, and assigning all tasks on date to make sure there are no other related bugs

### CODE REVIEW

File: packages/api/src/controllers/taskController.js [hunk 1]
```diff
@@ -94,17 +69,43 @@ const taskController = {
...
+          newAssigneeUserId,
+          oldAssigneeUserId,
+          currentTaskTranslationKey,
+          farm_id,
+          user_id,
+        );
+      }
+
+      // assign all other unassigned tasks due on this day to newAssigneeUserId
       const available_tasks = await TaskModel.query()
```
Comment (Cali0707) [inline packages/api/src/controllers/taskController.js:101]: Can you refactor this call into the model?

File: packages/api/src/controllers/taskController.js [hunk 2]
```diff
@@ -114,21 +115,25 @@ const taskController = {
       const result = await TaskModel.query()
```
Comment (Cali0707) [inline packages/api/src/controllers/taskController.js:115]: Can this one also be refactored into the model?

**Review by Cali0707 — CHANGES_REQUESTED** [review]
These changes work for me - just requesting two small refactors to improve the code quality.

**Comment by fsayyid** [comment]
I refactored those two queries into the task model.

**Review by Cali0707 — APPROVED** [review]
Looks great!

**Review by smattingly — APPROVED** [review]
Very well done!

**Comment by smattingly** [comment]
@fsayyid I'm a little unclear on how this (doesn't) relate to LF2323 or other tickets. Could you please update Jira ticket(s) status as needed due to this merge? Thanks.

**Comment by fsayyid** [comment]
@smattingly  I noticed this issue while working on LF-2323, and rather than creating a separate issue, me and Kevin decided to add it as a part of LF-2323. I will move LF-2323 into verification.

**Pull request merged**